### PR TITLE
No screenshot when AssumptionViolatedException occurs

### PR DESCRIPTION
**Pull request by theJNOVAK**
issue: #1506

### CODE REVIEW

**Comment by mlabuda** [comment]
You exclude screenshots when AVE is thrown only in Befores. There are many other places where exceptions are thrown and thus those places have to be patched as well. E.g. extensions, afters, method executions. Fix it please

**Comment by theJNOVAK** [comment]
Checks added to other suitable places.

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIAfterClassExtensions.java [hunk 1]
```diff
@@ -74,7 +74,9 @@ public void evaluate() throws Throwable {
 				}
 			} catch (Throwable e) {
 				log.error("Run method runAfterTestClass() of class " + after.getClass().getCanonicalName() + " failed", e);
```
Comment (mlabuda) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIAfterClassExtensions.java]: move error log to condition body

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIAfterTestExtensions.java [hunk 2]
```diff
@@ -75,7 +75,9 @@ public void evaluate() throws Throwable {
 				}
 			} catch (Throwable e) {
 				log.error("Run method runAfterTest() of class " + after.getClass().getCanonicalName() + " failed", e);
```
Comment (mlabuda) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIAfterTestExtensions.java]: move error log to condition body

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIBeforeClassExtensions.java [hunk 3]
```diff
@@ -65,7 +65,9 @@ public void evaluate() throws Throwable {
 			}
 		} catch (Throwable e) {
 			log.error("Run method runBeforeTestClass() of class " + before.getClass().getCanonicalName() + " failed", e);
```
Comment (mlabuda) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIBeforeClassExtensions.java]: move error log to condition body

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIBeforeTestExtensions.java [hunk 4]
```diff
@@ -66,7 +66,9 @@ public void evaluate() throws Throwable {
 			}
 		} catch (Throwable e) {
 			log.error("Run method runBeforeTest() of class " + before.getClass().getCanonicalName() + " failed", e);
```
Comment (mlabuda) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIBeforeTestExtensions.java]: move error log to condition body

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/AbstractStatementWithScreenshot.java [hunk 5]
```diff
@@ -104,6 +105,16 @@ protected void createScreenshot(String description, Class<?> extensionClass) {
...
 	}
 	
+	/**
+	 * Tests if screenshot is required on throwable.
+	 * 
+	 * @param throwable the throwable
+	 * @return true if required, false otherwise
+	 */
+	protected boolean isScreenshotRequired(Throwable throwable){
+		return !(throwable instanceof AssumptionViolatedException);
```
Comment (mlabuda) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/AbstractStatementWithScreenshot.java]: use check of exception type in Requirements class as well - there is a logic in canFulfill and fulfil methods where exception is catched and screenshot captured. We should have it consistent.

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIAfterClassExtensions.java (hunk 1 above)
Comment (theJNOVAK) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIAfterClassExtensions.java]: moved

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIAfterTestExtensions.java (hunk 2 above)
Comment (theJNOVAK) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIAfterTestExtensions.java]: moved

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIBeforeClassExtensions.java (hunk 3 above)
Comment (theJNOVAK) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIBeforeClassExtensions.java]: moved

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIBeforeTestExtensions.java (hunk 4 above)
Comment (theJNOVAK) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/RunIBeforeTestExtensions.java]: moved

File: plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/AbstractStatementWithScreenshot.java (hunk 5 above)
Comment (theJNOVAK) [inline plugins/org.jboss.reddeer.junit/src/org/jboss/reddeer/junit/internal/runner/statement/AbstractStatementWithScreenshot.java]: Check added to all screenshot captures in the class.

**Comment by mlabuda** [comment]
testPR

**Comment by mlabuda** [comment]
Please rebase on master. Run locally via mvn clean install -DskipTests to ensure there are no javadoc issues. Then push to the branch.

**Comment by odockal** [comment]
@mlabuda You must add javadoc:aggregate to yout maven commnad to test javadoc.

**Comment by theJNOVAK** [comment]
Branch rebased on upstream/master. Javadoc checked.

**Comment by rawagner** [comment]
testPR

**Pull request merged**